shopify_python/google_styleguide.py

```diff
@@ -251,3 +256,48 @@ def __use_cond_expr(self, node):  # type: (astroid.If) -> None
                 else_body_name = node.orelse[0].targets[0].name
                 if if_body_name == else_body_name:
                     self.add_message('cond-expr', node=node)
+
+    def __lambda_func(self, node):  # type: (astroid.Lambda) -> None
+        """Prefer Operator Function to Lambda Functions"""
+
+        if isinstance(node.body, astroid.UnaryOp):
+            unary_operators = {
+                "~": "invert",
+                "-": "neg",
+                "+": "pos"
+            }
+            operator = unary_operators.get(node.body.op)
+            varname = node.body.operand.name
+            lamfun = "lambda " + varname + ": " + node.body.op + " " + varname
+            opfun = "Operator." + operator
+            self.add_message('lambda-func', node=node, args={'op': opfun, 'lamfun': lamfun})
+        elif isinstance(node.body, astroid.BinOp):
+            self.__traverse_lambda(node.body)
+
+    def __traverse_lambda(self, node):
+        if shopify_python.ast.count_tree_size(node) == 3:
+            binary_operators = {
+                "+": "add",
+                "-": "sub",
+                "*": "mul",
+                "/": "truediv",
+                "**": "pow",
+                "%": "modulo",
+                "<": "lt",
+                "<=": "le",
+                "==": "eq",
+                "!=": "ne",
+                ">=": "ge",
+                ">": "gt"
+            }
+            operator = binary_operators.get(node.op)
+            if operator:
+                left = node.left.name
+                right = node.right.name
+                lamfun = "lambda " + left + ', ' + right + ": " + " ".join([left, node.op, right])
+                opfun = "Operator." + operator
+                self.add_message('lambda-func', node=node, args={'op': opfun, 'lamfun': lamfun})
+            return
+        elif shopify_python.ast.count_tree_size(node) > 3:
```

Comment:
> For common operations like multiplication, use the functions from the operator module instead of lambda functions. For example, prefer operator.mul to lambda x, y: x * y.

`lambda a, b: a * b` is definitely harder to read than `operator.mul` when used in `reduce`, e.g.
```
reduce(lambda a, b: a * b, integers)
reduce(operator.mul, integers)
```

But for larger expressions, I'd argue that `lambda a, b, c: a * b + c` is easier to read than  `lambda a, b, c: operator.mul(a, operator.add(b, c))`, but it's debatable e.g.
```
c = 3
reduce(lambda a, b: a * b + c, integers)
reduce(lambda a, b: operator.mul(a, operator.add(b, c)), integers)
```

@erikwright @honkfestival @JasonMWhite thoughts?